crates/repl_cli/src/lib.rs

```diff
@@ -37,10 +41,25 @@ pub struct ReplHelper {
     state: ReplState,
 }
 
+static BACKTRACE: OnceLock<Arc<Mutex<Option<String>>>> = OnceLock::new();
+
+fn init_backtrace_storage() {
+    BACKTRACE.get_or_init(|| Arc::new(Mutex::new(None)));
+}
+
+fn panic_hook(_: &PanicInfo) {
+    if let Some(storage) = BACKTRACE.get() {
+        *storage.lock().unwrap() = Some(Backtrace::force_capture().to_string());
```

Comment:
Rather than unwrap here, what about silently doing nothing if the lock is poisoned? That will avoid a double-panic in that situation, which I've found to be frustrating to debug.

Reply:
i don't understand how the lock could ever get poisoned but I don't feel particularly strongly about this, seems reasonable